pydantic/_internal/_generate_schema.py

```diff
@@ -1039,7 +1038,9 @@ def _match_generic_type(self, obj: Any, origin: Any) -> CoreSchema:  # noqa: C90
         if from_property is not None:
             return from_property
 
-        if _typing_extra.origin_is_union(origin):
+        if isinstance(origin, TypeAliasType):
+            return self._type_alias_type_schema(obj)
```

Comment:
Why was it moved?

Reply:
I think without a special reason (e.g. `dataclass` below), these matching clauses should better be put together. Looks like when `origin` is of type `TypeAliasType`, `from_property` is always `None` (at least in the current tests), so moving it does not have any actual effect.